### PR TITLE
Adopt Carbon spacing system

### PR DESCRIPTION
**Pull request by delapuente**
This PR follows [Carbon spacing guidelines](https://www.carbondesignsystem.com/guidelines/spacing) for adjusting margins and paddings. It tries to use the layout scale for margins and the spacing scale for paddings and organizing component hierarchies, when possible.

It does not use spacing, nor layout scale token in borders or image sizes.

### CODE REVIEW

**Comment by delapuente** [comment]
Mostly agree on everything. I'll review them individually to identify those where the scale does not reach the correct value or where the component is being used as a convenient group of components (as was the case of the hero moment discussed offline).

File: pages/experiments/index.vue [hunk 1]
```diff
@@ -91,10 +91,10 @@ main {
 #featured-experiments {
   color: $inverse-01;
   background-color: $inverse-02;
-  padding-bottom: 7rem;
+  padding-bottom: $layout-06;
 
   h2 {
-    margin-bottom: 4rem;
+    margin-bottom: $layout-05;
```
[the remark this answers is not in the document]
Comment (delapuente) [reply pages/experiments/index.vue:97]: I use `#featured-experiments` element like a group (more than like a component) to determine the layout of the index page constrained to this section.

File: pages/advocates/index.vue [hunk 2]
```diff
@@ -198,7 +198,7 @@ main {
   background-color: $inverse-02;
 
   ol {
-    margin-top: 1rem;
+    margin-top: $layout-01;
```
[the remark this answers is not in the document]
Comment (y4izus) [reply pages/advocates/index.vue:201]: If 👍 means you agree, you forgot to replace it

File: components/ui/footer/FooterLink.vue [hunk 3]
```diff
@@ -44,7 +44,7 @@ export default class extends Vue {
   text-decoration: none;
   display: inline-block;
   width: 100%;
-  margin-top: .5rem;
+  margin-top: $spacing-03;
```
[the remark this answers is not in the document]
Comment (delapuente) [reply components/ui/footer/FooterLink.vue:47]: I'll try, the `layout` scale is not that subtle and I was treating the subcomponents of the footer component as integral parts of the component and so, using the `spacing` scale made more sense to me. Non a strong argument though.

**Comment by delapuente** [comment]
It looks not bad at all. Can you review me again?

**Comment by y4izus** [comment]
It's ok for me, merging

**Pull request merged**